src/pages/library/common.js

```diff
@@ -58,11 +61,158 @@ const computeLabels = (allLabels, selectedLabels) => _.flow(
+        _.map(index => h(Link, {
+          style: {
+            fontWeight: 'bold', display: 'flex', justifyContent: 'center', alignItems: 'center', padding: 5, borderRadius: '50%', minWidth: 20,
+            background: filteredBy === String.fromCharCode(index) ? colors.accent(0.3) : 'transparent'
+          },
+          'aria-label': filteredBy === String.fromCharCode(index) ? `Filtering by ${String.fromCharCode(index)}` : `Filter option: ${String.fromCharCode(index)}`,
+          onClick: () => {
+            const charFromCode = String.fromCharCode(index)
+            setFilteredBy(charFromCode)
+            onFilterByLetter(charFromCode)
+          }
+        }, [String.fromCharCode(index)]), _.range(65, 65 + 26))
+      ),
+      h(Link, {
+        onClick: () => {
+          setFilteredBy()
+          setFilterType(filterOptions.input)
+          onClear()
+        }, 'aria-label': 'Search by text input'
+      }, [
+        span({ className: 'fa-stack fa-2x' }, [
+          icon('circle', { size: 40, className: 'fa-stack-2x', style: { color: colors.primary('light'), opacity: 0.2 } }),
+          icon('search', { size: 20, className: 'fa-stack-1x', style: { color: colors.primary('light') } })
+        ])
+      ])
+    ]),
+    filterType === filterOptions.alpha && filteredBy && div({ style: { display: 'flex', flexDirection: 'row', justifyContent: 'flex-end', marginTop: 5 } }, [
+      h(Link, {
+        style: { fontSize: '1rem' },
+        onClick: () => {
+          setFilteredBy()
+          onClear()
+        }
+      }, ['Clear filter'])
+    ]),
+    filterType === filterOptions.input && div({ style: { display: 'flex', alignItems: 'center', flexDirection: 'row' } }, [
+      h(DelayedSearchInput, {
+        style: { borderRadius: 25, borderColor: colors.dark(0.2), width: '100%', maxWidth: 575, height: '3rem', marginRight: 20 },
+        debounceMs: 25,
+        value: filterSearchText,
+        'aria-label': `Search for ${name} filter options`,
+        placeholder: 'Search keyword',
+        icon: 'search',
+        onChange: searchText => {
+          setFilterSearchText(searchText)
+          onFilterBySearchText(searchText)
+        }
+      }),
+      h(Link, {
+        style: { fontSize: '1rem' }, onClick: () => {
+          setFilterSearchText('')
+          setFilterType(filterOptions.alpha)
+          onClear()
+        }
+      }, ['Close search'])
+    ])
+  ])
+}
+
+const FilterModal = ({ name, labels, setShowAll, onTagFilter, listDataByTag, lowerSelectedTags }) => {
+  // Filter Modal Vars
+  const filteredTags = (_.flow(
+    _.map(label => {
+      const lowerTag = _.toLower(label)
+      return [lowerTag, listDataByTag[lowerTag]]
+    }),
+    _.fromPairs
+  )(labels))
+  const labelsByFirstChar = _.groupBy(label => _.capitalize(label)[0], labels)
+  const [filterChanges, setFilterChanges] = useState({})
+  const [filteredLabels, setFilteredLabels] = useState(labels)
+
+  return h(Modal, {
+    title: ``,
```

Comment:
I think for the moment we need to put the title in here. The reason being, title uses an `aria-labelledby` for the title and it is not rendering the associated element.

The title can be a component so we could the following to match the styling:
```suggestion
    title: div({ style: { fontSize: '1.325rem', fontWeight: 700, marginTop: 0 } }, [`Filter by: "${name}"`]),
```
I discovered this issue using the [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) tool/bookmarklet